src/goto-instrument/contracts/doc/user/contracts-loops.md

```diff
@@ -5,16 +5,23 @@ Back to @ref contracts-user
 @tableofcontents
 
 CBMC offers support for loop contracts, which includes four basic clauses:
+- an _assigns_ clause for declaring the memory locations assignable by the loop,
 - an _invariant_ clause for establishing safety properties,
 - a _decreases_ clause for establishing termination,
-- an _assigns_ clause for declaring the memory locations assignable by the loop,
 - a _frees_ clause for declaring the pointers freeable by the loop.
+**The four clauses need to be declared in this sequence to avoid errors. Each loop contract should only have one assign clause that contains all assign targets.**
+Loop contract is off by default. To enable CBMC to check for loop contracts, add the `--apply-loop-contract` flags at the `goto-instrument` step.
 
-These clauses formally describe an abstraction of a loop for the purpose of a proof.
+These clauses formally describe an abstraction of a loop for the purpose of an unbounded proof.
 CBMC also provides a series of built-in constructs
 to aid writing loop contracts (e.g., _history variables_ and _quantifiers_).
+Note for the content of the loop, only the invariants specified in the loop contract 
+will be verified, so any properties of interest should be explicitely declared in the 
+loop invariant clauses.
```

Comment:
I'm not sure I understand what you mean by "properties of interest should be explicitly declared" -- what is a "property" here, and how does one "declare" it?

Reply:
By properties of interest I mean the program behaviors that the developer is interested in verifying, and by declaring it I mean specifying these expected behaviors in the loop contract clauses.

Reply:
I would say "CBMC will prove if the loop contracts describes a sound abstraction of the loop, and use the abstraction in place of the loop."

Reply:
Revised!